From b00d66ba2c3fc1c2ef23e29872941e07d97a0cca Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 30 Nov 2020 00:10:55 +0200
Subject: [PATCH] Revert "[SW-25120] gaudi: security: utilize sp sram"

This reverts commit ce21f7812fb8f6c4c3b317a6b3c1bef833b02188.

This commit is replaced by boot info desc

Change-Id: I3c95d7bac0adb65f103923e440b267e472b15389
Signed-off-by: Guy Nisan <gnisan@habana.ai>
(cherry picked from commit 209be9e8e2d3c322adb47d2031f47e9854928957)
---

diff --git a/gaudi/security.c b/gaudi/security.c
index eb2861f..fe84029 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -54,9 +54,6 @@
 #include <spi.h>
 #include <sha256.h>
 
-/* Scratchpad SRAM base address */
-#define HL_SP_SRAM_BASE		0x7FFBFE0000
-
 #define PROT_REGS_COUNT		32
 #define PROT_REGS_MAX		(PROT_REGS_COUNT - 1) /* Secure config -> LKD */
 #define BITS_IN_REG		32
@@ -105,9 +102,6 @@
 	.is_invalid = true,
 };
 
-static struct hl_sec_sp_sram *preboot_security_sp_sram =
-	(struct hl_sec_sp_sram *)(HL_SP_SRAM_BASE + SEC_SP_SRAM_OFFSET);
-
 static const struct hl_sec_otp *preboot_security_get_otp(void)
 {
 	struct hl_sec_otp *otp = &preboot_security_otp_cache.otp;
@@ -131,13 +125,13 @@
 	return otp;
 }
 
-#ifdef FUSE_SUPPORT
 static const struct hl_sec_efuse *preboot_security_get_fuse(void)
 {
 	struct hl_sec_efuse *fuse;
 	u32 *val;
 	int i;
 
+#ifdef FUSE_SUPPORT
 	if (preboot_security_fuse_cache.is_invalid) {
 		fuse = &preboot_security_fuse_cache.fuse;
 
@@ -148,24 +142,8 @@
 		}
 		preboot_security_fuse_cache.is_invalid = false;
 	}
-
-	return &preboot_security_fuse_cache.fuse;
-}
-
-static void preboot_security_sp_sram_write_efuse(void)
-{
-	/* Do nothing for real eFuse */
-}
-
-#else /* FUSE_SUPPORT */
-
+#else
 #define FAKE_FUSE_OFFSET ((2 * 1024 * 1024) + (64 * 1024))
-static const struct hl_sec_efuse *preboot_security_get_fuse(void)
-{
-	struct hl_sec_efuse *fuse;
-	u32 *val;
-	int i;
-
 	if (preboot_security_fuse_cache.is_invalid) {
 		fuse = &preboot_security_fuse_cache.fuse;
 
@@ -181,50 +159,9 @@
 
 		preboot_security_fuse_cache.is_invalid = false;
 	}
+#endif
 
 	return &preboot_security_fuse_cache.fuse;
-}
-
-static void preboot_security_sp_sram_write_efuse(void)
-{
-	const struct hl_sec_efuse *efuse;
-	int i;
-
-	efuse = preboot_security_get_fuse();
-	if (!efuse)
-		return;
-
-	for (i = 0; i < ARRAY_SIZE(efuse->data); i++)
-		preboot_security_sp_sram->efuse.data[i] = efuse->data[i];
-
-	preboot_security_sp_sram->efuse_header.magic = SEC_SP_SRAM_MAGIC;
-}
-#endif /* FUSE_SUPPORT */
-
-static void preboot_security_sp_sram_write_otp(void)
-{
-	const struct hl_sec_otp *otp;
-	int i;
-
-	otp = preboot_security_get_otp();
-	if (!otp)
-		return;
-
-	for (i = 0; i < ARRAY_SIZE(otp->data); i++)
-		preboot_security_sp_sram->otp.data[i] = otp->data[i];
-
-	preboot_security_sp_sram->otp_header.magic = SEC_SP_SRAM_MAGIC;
-}
-
-static void preboot_security_sp_sram_init(void)
-{
-	int i;
-
-	for (i = 0; i < ARRAY_SIZE(preboot_security_sp_sram->data); i++)
-		preboot_security_sp_sram->data[i] = 0;
-
-	preboot_security_sp_sram_write_otp();
-	preboot_security_sp_sram_write_efuse();
 }
 
 bool preboot_security_key_is_valid(int key_idx, const void *pub_key,
@@ -1404,9 +1341,6 @@
 
 	cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_NOT_RDY, true);
 	preboot_puts("Security init started...\n");
-
-	/* Initiate SP SRAM contained security data */
-	preboot_security_sp_sram_init();
 
 	is_sec_en = preboot_security_is_enabled();
 	if (is_sec_en) {
